src/service/vsync-impl.js

```diff
@@ -444,6 +469,28 @@ export class Vsync {
       this.win.setTimeout(fn, timeToCall);
     };
   }
+
+  /**
+   * @param {!Array<!MutationRecord>|null} records
+   * @private
+   */
+  mutationHandler_(records) {
+    if (!records) {
+      return;
+    }
+    for (let i = 0; i < records.length; i++) {
+      const {target, type} = records[i];
+      dev().error('VSYNC', `Detected ${type} mutation on ${target.tagName} ` +
+        'outside of mutation phase: %s', target);
```

Comment:
Let's give engs a hint on what they ought to do e.g. use `BaseElement.mutateElement()`.